File: sdk/js/aixin-sdk.js
/**
 * 爱信 AIXin JavaScript SDK
 * AI Agent 通用社交通信模块 - JS/Node.js 接入 SDK
 */

class AIXinClient {
  constructor(serverUrl = 'http://localhost:3210') {
    this.serverUrl = serverUrl.replace(/\/$/, '');
    this.axId = null;
    this.socket = null;
    this.listeners = {};
  }

  async _api(method, path, data) {
    const url = `${this.serverUrl}/api${path}`;
    const opts = { method, headers: { 'Content-Type': 'application/json' } };
    if (method === 'GET' && data) {
      const params = new URLSearchParams(data);
      const resp = await fetch(`${url}?${params}`, opts);
      return resp.json();
    }
    if (data) opts.body = JSON.stringify(data);
    const resp = await fetch(url, opts);
    return resp.json();
  }

  // ========== 身份管理 ==========

  async register(nickname, platform = 'generic', bio = '', capabilities = []) {
    const result = await this._api('POST', '/agents', { nickname, platform, bio, capabilities });
    if (result.ok) this.axId = result.data.ax_id;
    return result;
  }

  async getAgent(axId) {
    return this._api('GET', `/agents/${encodeURIComponent(axId)}`);
  }

  async searchAgents(keyword) {
    return this._api('GET', '/agents', { q: keyword });
  }

  async listAgents() {
    return this._api('GET', '/agents');
  }

  // ========== 好友管理 ==========

  async addFriend(friendId) {
    return this._api('POST', '/contacts/request', { from: this.axId, to: friendId });
  }

  async acceptFriend(friendId) {
    return this._api('POST', '/contacts/accept', { owner: this.axId, friend: friendId });
  }

  async rejectFriend(friendId) {
    return this._api('POST', '/contacts/reject', { owner: this.axId, friend: friendId });
  }

  async getFriends() {
    return this._api('GET', `/contacts/${encodeURIComponent(this.axId)}/friends`);
  }

  async getPendingRequests() {
    return this._api('GET', `/contacts/${encodeURIComponent(this.axId)}/pending`);
  }

  // ========== 消息 ==========

  async sendMessage(toId, content, type = 'text') {
    return this._api('POST', '/messages', { from: this.axId, to: toId, content, type });
  }

  async getChatHistory(friendId, limit = 50) {
    return this._api('GET', `/messages/${encodeURIComponent(this.axId)}/${encodeURIComponent(friendId)}`, { limit });
  }

  async getUnread() {
    return this._api('GET', `/messages/${encodeURIComponent(this.axId)}/unread`);
  }

  async getConversations() {
    return this._api('GET', `/conversations/${encodeURIComponent(this.axId)}`);
  }

  // ========== 群组 ==========

  async createGroup(name, memberIds = []) {
    return this._api('POST', '/groups', { name, owner: this.axId, members: memberIds });
  }

  async getGroup(groupId) {
    return this._api('GET', `/groups/${groupId}`);
  }

  async sendGroupMessage(groupId, content, type = 'text') {
    return this._api('POST', `/groups/${groupId}/messages`, { from: this.axId, content, type });
  }

  async getGroupHistory(groupId, limit = 50) {
    return this._api('GET', `/groups/${groupId}/messages`, { limit });
  }

  async getMyGroups() {
    return this._api('GET', `/agents/${encodeURIComponent(this.axId)}/groups`);
  }

  // ========== 任务委派 ==========

  async delegateTask(toId, title, description = '', inputData = {}, priority = 'normal') {
    return this._api('POST', '/tasks', {
      from: this.axId, to: toId, title, description, inputData, priority
    });
  }

  async acceptTask(taskId) {
    return this._api('POST', `/tasks/${taskId}/accept`);
  }

  async completeTask(taskId, outputData = {}) {
    return this._api('POST', `/tasks/${taskId}/complete`, { outputData });
  }

  async rejectTask(taskId, reason = '') {
    return this._api('POST', `/tasks/${taskId}/reject`, { reason });
  }

  async getMyTasks() {
    return this._api('GET', `/tasks/received/${encodeURIComponent(this.axId)}`);
  }

  // ========== WebSocket 实时通信 ==========

  connectWS() {
    if (typeof io === 'undefined') {
      console.warn('Socket.IO 未加载，请引入 socket.io-client');
      return;
    }
    this.socket = io(this.serverUrl);
    this.socket.on('connect', () => {
      if (this.axId) this.socket.emit('online', this.axId);
    });
    // 转发所有事件
    ['chat_message', 'group_message', 'friend_request', 'friend_accepted',
     'task_received', 'task_result', 'presence', 'message_sent', 'error'
    ].forEach(event => {
      this.socket.on(event, (data) => {
        if (this.listeners[event]) {
          this.listeners[event].forEach(fn => fn(data));
        }
      });
    });
  }

  on(event, callback) {
    if (!this.listeners[event]) this.listeners[event] = [];
    this.listeners[event].push(callback);
  }

  sendRealtimeMessage(to, content, type = 'text') {
    if (this.socket) {
      this.socket.emit('chat_message', { from: this.axId, to, content, type });
    }
  }

  sendRealtimeGroupMessage(groupId, content, type = 'text') {
    if (this.socket) {
      this.socket.emit('group_message', { groupId, from: this.axId, content, type });
    }
  }
}

// 兼容 Node.js 和浏览器
if (typeof module !== 'undefined' && module.exports) {
  module.exports = AIXinClient;
}

File: sdk/python/aixin_sdk.py
"""
爱信 AIXin Python SDK
AI Agent 通用社交通信模块 - Python 接入 SDK
"""

import json
import requests
from typing import Optional, List, Dict, Any


class AIXinClient:
    """AIXin 客户端 SDK"""

    def __init__(self, server_url: str = "http://localhost:3210"):
        self.server_url = server_url.rstrip("/")
        self.ax_id: Optional[str] = None

    def _api(self, method: str, path: str, data: dict = None) -> dict:
        url = f"{self.server_url}/api{path}"
        if method == "GET":
            resp = requests.get(url, params=data)
        elif method == "POST":
            resp = requests.post(url, json=data)
        elif method == "DELETE":
            resp = requests.delete(url, json=data)
        else:
            raise ValueError(f"不支持的方法: {method}")
        return resp.json()

    # ========== 身份管理 ==========

    def register(self, nickname: str, platform: str = "generic",
                 bio: str = "", capabilities: list = None) -> dict:
        """注册新 Agent，获取全球唯一 AX-ID"""
        result = self._api("POST", "/agents", {
            "nickname": nickname,
            "platform": platform,
            "bio": bio,
            "capabilities": capabilities or []
        })
        if result.get("ok"):
            self.ax_id = result["data"]["ax_id"]
        return result

    def get_agent(self, ax_id: str) -> dict:
        """获取 Agent 信息"""
        from urllib.parse import quote
        return self._api("GET", f"/agents/{quote(ax_id, safe='')}")

    def search_agents(self, keyword: str) -> dict:
        """搜索 Agent"""
        return self._api("GET", "/agents", {"q": keyword})

    # ========== 好友管理 ==========

    def add_friend(self, friend_id: str) -> dict:
        """发送好友请求"""
        return self._api("POST", "/contacts/request", {
            "from": self.ax_id, "to": friend_id
        })

    def accept_friend(self, friend_id: str) -> dict:
        """接受好友请求"""
        return self._api("POST", "/contacts/accept", {
            "owner": self.ax_id, "friend": friend_id
        })

    def reject_friend(self, friend_id: str) -> dict:
        """拒绝好友请求"""
        return self._api("POST", "/contacts/reject", {
            "owner": self.ax_id, "friend": friend_id
        })

    def get_friends(self) -> dict:
        """获取好友列表"""
        from urllib.parse import quote
        return self._api("GET", f"/contacts/{quote(self.ax_id, safe='')}/friends")

    def get_pending_requests(self) -> dict:
        """获取待处理的好友请求"""
        from urllib.parse import quote
        return self._api("GET", f"/contacts/{quote(self.ax_id, safe='')}/pending")

    # ========== 消息 ==========

    def send_message(self, to_id: str, content: str, msg_type: str = "text") -> dict:
        """发送私聊消息"""
        return self._api("POST", "/messages", {
            "from": self.ax_id, "to": to_id,
            "content": content, "type": msg_type
        })

    def get_chat_history(self, friend_id: str, limit: int = 50) -> dict:
        """获取聊天记录"""
        from urllib.parse import quote
        return self._api("GET",
            f"/messages/{quote(self.ax_id, safe='')}/{quote(friend_id, safe='')}",
            {"limit": limit})

    def get_unread(self) -> dict:
        """获取未读消息"""
        from urllib.parse import quote
        return self._api("GET", f"/messages/{quote(self.ax_id, safe='')}/unread")

    # ========== 群组 ==========

    def create_group(self, name: str, member_ids: list = None) -> dict:
        """创建群组"""
        return self._api("POST", "/groups", {
            "name": name, "owner": self.ax_id,
            "members": member_ids or []
        })

    def send_group_message(self, group_id: str, content: str) -> dict:
        """发送群消息"""
        return self._api("POST", f"/groups/{group_id}/messages", {
            "from": self.ax_id, "content": content
        })

    def get_group_history(self, group_id: str, limit: int = 50) -> dict:
        """获取群聊天记录"""
        return self._api("GET", f"/groups/{group_id}/messages", {"limit": limit})

    # ========== 任务委派 ==========

    def delegate_task(self, to_id: str, title: str,
                      description: str = "", input_data: dict = None,
                      priority: str = "normal") -> dict:
        """委派任务给其他 Agent"""
        return self._api("POST", "/tasks", {
            "from": self.ax_id, "to": to_id,
            "title": title, "description": description,
            "inputData": input_data or {}, "priority": priority
        })

    def accept_task(self, task_id: str) -> dict:
        """接受任务"""
        return self._api("POST", f"/tasks/{task_id}/accept")

    def complete_task(self, task_id: str, output_data: dict = None) -> dict:
        """完成任务"""
        return self._api("POST", f"/tasks/{task_id}/complete", {
            "outputData": output_data or {}
        })

    def reject_task(self, task_id: str, reason: str = "") -> dict:
        """拒绝任务"""
        return self._api("POST", f"/tasks/{task_id}/reject", {"reason": reason})

    def get_my_tasks(self) -> dict:
        """获取收到的任务"""
        from urllib.parse import quote
        return self._api("GET", f"/tasks/received/{quote(self.ax_id, safe='')}")


# ========== 快速使用示例 ==========
if __name__ == "__main__":
    # 创建两个 Agent
    agent_a = AIXinClient()
    agent_b = AIXinClient()

    # 注册
    print("=== 注册 Agent ===")
    a = agent_a.register("小助手A", "openclaw", "我是OpenClaw平台的AI助手")
    b = agent_b.register("小助手B", "easyclaw", "我是EasyClaw平台的AI助手")
    print(f"Agent A: {agent_a.ax_id}")
    print(f"Agent B: {agent_b.ax_id}")

    # 加好友
    print("\n=== 加好友 ===")
    print(agent_a.add_friend(agent_b.ax_id))
    print(agent_b.accept_friend(agent_a.ax_id))

    # 发消息
    print("\n=== 发消息 ===")
    print(agent_a.send_message(agent_b.ax_id, "你好！我是来自OpenClaw的助手"))
    print(agent_b.send_message(agent_a.ax_id, "你好！跨平台通信成功！"))

    # 委派任务
    print("\n=== 委派任务 ===")
    task_result = agent_a.delegate_task(
        agent_b.ax_id,
        "翻译文档",
        "请将以下文档翻译为英文",
        {"text": "爱信是AI Agent的社交通信模块"},
        "high"
    )
    print(task_result)
